test/parallel_api/numeric/numeric.ops/histogram.pass.cpp

```diff
@@ -0,0 +1,131 @@
+// -*- C++ -*-
+//===-- histogram.pass.cpp ------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(numeric)
+
+#include "support/utils.h"
+#include "support/histogram_serial_impl.h"
+
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+struct test_histogram_even_bins
+{
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size, typename T>
+    void
+    operator()(Policy&& exec, Iterator1 in_first, Iterator1 in_last, Iterator2 expected_bin_first,
+               Iterator2 expected_bin_last, Iterator3 bin_first, Iterator3 bin_last, Size n, T bin_min, T bin_max,
+               Size trash)
+    {
+        const Size bin_size = bin_last - bin_first;
+        histogram_sequential(in_first, in_last, bin_size, bin_min, bin_max, expected_bin_first);
+        auto orr = ::oneapi::dpl::histogram(exec, in_first, in_last, bin_size, bin_min, bin_max, bin_first);
```

Comment:
Where is defined `::oneapi::dpl::histogram` ?
Looks like it's defined in the file `include/oneapi/dpl/internal/histogram_extension_defs.h'
It's incorrect, because as far as I understand we shouldn't declare public algorithms inside internal folder.

Reply:
This is a good point... I followed the lead of the other extension algorithms `exclusive_scan_by_segment`, async routines, etc.

Do we need to add a new top level header like `include/oneapi/dpl/algorithm` for these extensions and another for  experimental APIs?  Right now they are included when one includes `algorithm` but defined in `include/oneapi/dpl/internal/`.

Reply:
I don't know exactly the right way, May be better to discuss with other colleagues.

Reply:
I think we need to consider it with the experimental async algorithms, and the other extensions:  

`include/oneapi/dpl/internal/binary_search_extension_defs.h`
`include/oneapi/dpl/internal/async_extension_defs.h` @reble 
`include/oneapi/dpl/internal/by_segment_extension_defs.h` @mmichel11 

@akukanov, @rarutyun: do you have an opinion here on where these extensions should be defined?

Reply:
After talking to @timmiesmith , he mentioned this was originally done this way to separate extensions what was being upstreamed to PSTL.  I think perhaps all of these extensions will end up in a compat or extensions folder eventually.